snuba/query/parser/functions.py

```diff
@@ -124,6 +132,26 @@ def parse_function(
         )
 
     name, args, alias = function_tuple
+
+    # If the first argument is a simple column name that refers to an array column
+    # (and we are not arrayJoining on that column, which would make it scalar again)
+    # we assume that the user actually means to check if any (or all) items in the
+    # array match the predicate, so we return an `any(x == value for x in array_column)`
+    # type expression. We assume that operators looking for a specific value (IN, =, LIKE)
+    # are looking for rows where any array value matches, and exclusionary operators
+    # (NOT IN, NOT LIKE, !=) are looking for rows where all elements match (eg. all NOT LIKE 'foo').
+    if name in FUNCTION_TO_OPERATOR:
+        if len(args) == 2 and args[0] in dataset_columns:
+            column = dataset_columns[args[0]]
+            if isinstance(column.type, Array) or (
+                isinstance(column.type, Materialized)
+                and isinstance(column.type.get_raw(), Array)
+            ):
```

Comment:
isn't `if isinstance(column.type.get_raw(), Array):` enough? 
I think you just need to check if the raw type excluded all the modifiers is an Array. If column.type is just Array, `column.type.get_raw()` would be `Array`

Reply:
No because `column` is sometimes a `FlattenedColumn` which doesn't have a `get_raw` function. So I have to check both types.

Reply:
The flattened column has a type which has a get_raw method. 
https://github.com/getsentry/snuba/blob/master/snuba/clickhouse/columns.py#L51

FlattenedColumn.type.get_raw(). 
That should always be there. Did you encounter any case where this was not true ?

Reply:
Ah I missed the extra `.type` call. I changed it.